app/lib/account/consent_backend.dart

```diff
@@ -182,17 +190,20 @@ class ConsentBackend {
     if (c == null) return null;
 
     // Checking that consent is for the current user.
-    if (c.userId != null && c.userId != user.userId) return null;
-    if (c.email != null && c.email != user.email) return null;
+    if (c.userIdOfConsent != null && c.userIdOfConsent != user.userId) {
+      return null;
+    }
+    if (c.email != null && c.email != user.email) {
+      return null;
+    }
     return c;
   }
 
   Future _accept(Consent consent) async {
     final action = _actions[consent.kind];
     await retry(
       () async {
-        await action?.onAccept(
-            consent.fromUserId, consent.userId, consent.args);
+        await action?.onAccept(consent);
```

Comment:
Unrelated, but wouldn't be healthy to handle the case where `action==null` explicitly, note: I think this is best in a follow up PR... or maybe I'm missing something. Can't we do the delete in a separate thing, so the intentions are clear?

Reply:
> Can't we do the delete in a separate thing, so the intentions are clear?

what do you mean by that?